added documentation for defclass docstring

digikar99 opened the pull request:
#1527 
The following works with bar.__doc__ returning "bar":
   
    (defclass bar [] "bar" [attr "value"]) 

while the following does not:

    (defclass bar [] [attr "value"] "bar")

(bar.\_\_doc__ evaluates to bar.\_\_doc__ itself.)

This seems to be the intended python-ic behaviour, if I am not wrong.

Kodiologist commented:
I think you forgot a trailing backtick. Did you try rendering the file?

Kodiologist commented:
Looks good, but the backtick fix should be in the first commit, and the underline fix should be in its own commit (if you want it in this PR). Try

    git reset HEAD~
    git commit --amend --patch
    git commit -a

brandonwillard commented on docs/language/api.rst:
```diff
@@ -415,13 +415,14 @@ They can be used to assign multiple variables at once:
 
-New classes are declared with ``defclass``. It can takes two optional parameters:
-a vector defining a possible super classes and another vector containing
+New classes are declared with ``defclass``. It can takes three optional parameters in the following order:
```
`takes` should be `take`.

brandonwillard commented on docs/language/api.rst:
```diff
@@ -415,13 +415,14 @@ They can be used to assign multiple variables at once:
-New classes are declared with ``defclass``. It can takes two optional parameters:
-a vector defining a possible super classes and another vector containing
+New classes are declared with ``defclass``. It can takes three optional parameters in the following order:
+a vector defining a possible super classes, a string (:term:`py:docstring`) and another vector containing
```
`a vector defining a possible super classes` should probably be `a list defining the super classes`.

digikar99 commented:
> I think you forgot a trailing backtick. Did you try rendering the file?

Yup, I had forgotten. I copied it from a few lines below, and didn't take note of the closing back-tick. Thanks for pointing that out.

No, I had not tried rendering. I will keep "build should be successful" in mind for the future pull requests.



> Looks good, but the backtick fix should be in the first commit, and the underline fix should be in its own commit (if you want it in this PR). Try
> 
> ```
> git reset HEAD~
> git commit --amend --patch
> git commit -a
> ```

I'll also keep "commit separately for each task, no matter how small that task is". Thanks for pointing out!



> I know you didn't introduce these mistakes, but, while you're changing those lines, this would be a great opportunity to fix them!

Sure, I have incorporated the suggested changes now. I have also added another attr-value example. 

Another good-first-issue I can think of is: fixing the missing syntax-highlighting (or more generally, fixing the warnings in Sphinx build process) for some of the documentation. Example sections include [`as->`](http://docs.hylang.org/en/stable/language/api.html#as). Should I create such an issue?

Kodiologist commented:
If you wish, sure.

Kodiologist commented:
Okay, I'll clean up the commit structure and merge.

Kodiologist commented:
@digikar99 I'm getting "Permission denied". Don't forget to turn on editing of your PR branches. (And in the future, make a new branch for each PR rather requesting a pull from `master`).

Kodiologist commented:
Ah, I got it. I was pushing to the wrong branch name because my local tracking branch wasn't named `master`.
